### PR TITLE
Add gmf-disclaimer in apps

### PR DESCRIPTION
**Pull request by adube**
This PR introduces the `gmf-disclaimer` directive in all GMF applications.

Note: this PR currently includes the content of #1341, since it hasn't been merged yet.

## Todo

 * [x] wait for #1341 to be merged
 * [x] review
 * [x] apply reviewer's corrections

## Live demos

 * http://adube.github.io/ngeo/gmf-disclaimer-in-apps/examples/contribs/gmf/apps/mobile/index.html
 * http://adube.github.io/ngeo/gmf-disclaimer-in-apps/examples/contribs/gmf/apps/desktop/index.html
 * http://adube.github.io/ngeo/gmf-disclaimer-in-apps/examples/contribs/gmf/apps/desktop_alt/index.html

### CODE REVIEW

**Comment by adube** [comment]
Ready for review.

**Comment by sbrunner** [comment]
Looks good for me :-)

**Comment by adube** [comment]
Thanks.  #1341 needs to be reviewed first, btw.

File: contribs/gmf/less/map.less [hunk 1]
```diff
@@ -102,3 +102,22 @@ button[ngeo-mobile-geolocation] {
...
+  z-index: 3;
+
+  .alert {
+    padding: 0.5rem 1rem;
+    margin: 0 0 0.5rem 0;
+  }
+
+  .alert-dismissable .close,
+  .alert-dismissible .close {
+    right: -5px;
```
Comment (sbrunner) [inline contribs/gmf/less/map.less]: why not in `rem`?

File: contribs/gmf/less/map.less [hunk 2]
```diff
@@ -102,3 +102,22 @@ button[ngeo-mobile-geolocation] {
...
+  }
+}
```
Comment (sbrunner) [inline contribs/gmf/less/map.less]: missing end line :-)

File: contribs/gmf/less/mobile.less [hunk 3]
```diff
@@ -125,3 +125,13 @@ button[ngeo-mobile-geolocation] {
...
+
+/** Disclaimer */
+main gmf-disclaimer {
+  text-align: left;
+
+  button {
+    height: 1.5rem;
+    width: 1.5rem;
+  }
+}
```
Comment (sbrunner) [inline contribs/gmf/less/mobile.less]: missing end line :-)

File: contribs/gmf/apps/desktop_alt/index.html [hunk 4]
```diff
@@ -110,6 +110,9 @@
           </gmf-backgroundlayerselector>
         </div>
         <gmf-map class="gmf-map" gmf-map-map="mainCtrl.map"></gmf-map>
+        <gmf-disclaimer
+          gmf-disclaimer-map="::mainCtrl.map">
+        </gmf-disclaimer>
```
Comment (sbrunner) [inline contribs/gmf/apps/desktop_alt/index.html]: Can be done later but in the  desktop_alt we will have the modal disclamer

File: contribs/gmf/less/map.less (hunk 1 above)
Comment (ger-benjamin) [inline contribs/gmf/less/map.less]: Please don't use px in app style files

File: contribs/gmf/less/map.less [hunk 5]
```diff
@@ -102,3 +102,22 @@ button[ngeo-mobile-geolocation] {
...
   }
 }
+
+/** Disclaimer */
+gmf-disclaimer {
+  position: absolute;
+  bottom: 0.5rem;
+  left: 0.5rem;
+  width: 30rem;
+  z-index: 3;
```
Comment (ger-benjamin) [inline contribs/gmf/less/map.less]: Perhaps you can use z-level vars here

File: contribs/gmf/less/map.less [hunk 6]
```diff
@@ -102,3 +102,22 @@ button[ngeo-mobile-geolocation] {
     clear: left;
   }
 }
+
+/** Disclaimer */
+gmf-disclaimer {
+  position: absolute;
+  bottom: 0.5rem;
+  left: 0.5rem;
```
Comment (ger-benjamin) [inline contribs/gmf/less/map.less]: @half-app-margin is better here

File: contribs/gmf/less/map.less [hunk 7]
```diff
@@ -102,3 +102,22 @@ button[ngeo-mobile-geolocation] {
...
+  width: 30rem;
+  z-index: 3;
+
+  .alert {
+    padding: 0.5rem 1rem;
```
Comment (ger-benjamin) [inline contribs/gmf/less/map.less]: @half-app-margin and @app-margin is better here

**Comment by ger-benjamin** [comment]
And I think that the disclaimer must be aligned  like the zoom buttons on the left of the application(or menu button on mobile) and same distance for the bottom (twice `@app-margin I guess`).

**Comment by adube** [comment]
Corrections applied.  Ready for merge.

**Comment by adube** [comment]
Rebased onto master.

File: contribs/gmf/less/map.less [hunk 8]
```diff
@@ -102,3 +102,22 @@ button[ngeo-mobile-geolocation] {
...
+gmf-disclaimer {
+  position: absolute;
+  bottom: @half-app-margin;
+  left: @app-margin;
+  width: 30rem;
+  z-index: @above-content-index;
+
+  .alert {
+    padding: @half-app-margin @app-margin;
+    margin: 0 0@half-app-margin  0;
```
Comment (ger-benjamin) [inline contribs/gmf/less/map.less]: spacing problem (misplaced)

Comment (adube) [inline contribs/gmf/less/map.less]: I fixed it already. I simply didn't update the live demo.  I'll do this right away.

**Comment by adube** [comment]
Demos updated.

**Comment by ger-benjamin** [comment]
I don't know what to do with the covering of the disclaimers over the infobar and the background selector.
But for me that can be modified in another PR. This one is ready to merge.
Ok @sbrunner ?

**Comment by adube** [comment]
I agree with deciding what to do with this in an other PR.

**Comment by sbrunner** [comment]
Thanks :-)

**Pull request merged**